apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/Logging/ILoggerExtensionsTest.cs

```diff
@@ -0,0 +1,211 @@
+            var augmentedLogger = originalLogger.WithLabels(new KeyValuePair<string, string>("job_title", "hairdresser"));
+
+            augmentedLogger.LogInformation("Cutting hair");
+
+            scopeMock.Verify(scope => scope.Dispose(), Times.Once());
+            loggerMock
+                .VerifyLog(Times.Once())
+                .VerifyBeginScope(Times.Once())
+                .VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public void WithLabels_UseOriginalAfterAugmented()
+        {
+            var scopeMock = GetScopeMock();
+
+            var loggerMock = new Mock<ILogger>(MockBehavior.Strict)
+                .SetupLog("Cutting hair")
+                .SetupBeginScope(scopeMock.Object, new KeyValuePair<string, string>("job_title", "hairdresser"));
+
+            var originalLogger = loggerMock.Object;
+            var augmentedLogger = originalLogger.WithLabels(new KeyValuePair<string, string>("job_title", "hairdresser"));
+
+            augmentedLogger.LogInformation("Cutting hair");
+            scopeMock.Verify(scope => scope.Dispose(), Times.Once());
+            originalLogger.LogInformation("Cutting hair");
+
+            loggerMock
+                .VerifyLog(Times.Exactly(2))
+                .VerifyBeginScope(Times.Once())
+                .VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public void WithAddedLabels()
+        {
+            var scopeMock = GetScopeMock();
+
+            var loggerMock = new Mock<ILogger>(MockBehavior.Strict)
+                .SetupLog("Cutting hair")
+                .SetupBeginScope(scopeMock.Object,
+                    new KeyValuePair<string, string>("job_title", "hairdresser"),
+                    new KeyValuePair<string, string>("tool", "scissors"));
+
+            var originalLogger = loggerMock.Object;
+            var oneLabelLogger = originalLogger.WithLabels(new KeyValuePair<string, string>("job_title", "hairdresser"));
+            var twoLabelLogger = oneLabelLogger.WithAddedLabels(new KeyValuePair<string, string>("tool", "scissors"));
+
+            Assert.NotSame(oneLabelLogger, twoLabelLogger);
+
+            twoLabelLogger.LogInformation("Cutting hair");
+
+            scopeMock.Verify(scope => scope.Dispose(), Times.Once());
+
+            loggerMock
+                .VerifyLog(Times.Once())
+                .VerifyBeginScope(Times.Once())
+                .VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public void WithAddedLabels_NoWithLabels()
+        {
+            var scopeMock = GetScopeMock();
+
+            var loggerMock = new Mock<ILogger>(MockBehavior.Strict)
+                .SetupLog("Cutting hair")
+                .SetupBeginScope(scopeMock.Object, new KeyValuePair<string, string>("job_title", "hairdresser"));
+
+            var originalLogger = loggerMock.Object;
+            var augmentedLogger = originalLogger.WithAddedLabels(new KeyValuePair<string, string>("job_title", "hairdresser"));
```

Comment:
Nit: use `KeyValuePair.Create` instead of `new KeyValuePair<string, string>` everywhere in this file, just for brevity?

Reply:
It's not available on .net461 :(